src/js/components/ServicePlanProgressBar.js

```diff
@@ -0,0 +1,92 @@
+import classNames from 'classnames';
+import React from 'react';
+
+import ServicePlan from '../structs/ServicePlan';
+
+const labelMap = {
+  'InProgress': 'Updating'
+};
+
+class ServicePlanProgressBar extends React.Component {
+  getPhaseLabel(servicePlan) {
+    let phases = servicePlan.getPhases();
+
+    return (
+      <span className="text-overflow-break-word">
+        Phase {phases.getActiveIndex()} of {phases.getItems().length}:
+        {` ${phases.getActive().getName()} `}
+        <a className="clickable" onClick={this.props.onViewDetailsClick}>
+          View Details
+        </a>
+      </span>
+    );
+  }
+
+  getStatusLabel(servicePlan) {
+    let status = servicePlan.getStatus();
+
+    return labelMap[status] || status;
+  }
+
+  render() {
+    let {props} = this;
+    let {servicePlan} = props;
+
+    let fillClasses = classNames('progress-bar-fill',
```

Comment:
not gonna lie, i think this is the right way to go, but aesthetically isn't the greatest :(